### PR TITLE
do not use secret store to check whether user is logged in in engine

### PR DESCRIPTION
**Pull request by heronhaye**

### CODE REVIEW

File: go/libkb/active_device.go [hunk 1]
```diff
@@ -225,6 +225,7 @@ func (a *ActiveDevice) clear() (KeychainMode, error) {
 
 	a.uv = keybase1.UserVersion{}
 	a.deviceID = ""
+	a.deviceName = ""
```
Comment (heronhaye) [inline go/libkb/active_device.go]: small bugfix. i noticed this wasn't being cleared

**Comment by jakob223** [comment]
cc @patrickxb

**Comment by heronhaye** [comment]
Idk if this is the whole story. For example if you restart then you'll be "Logged in" again.

**Comment by heronhaye** [comment]
But yes @jakob223 as you hypothesized, "isLoggedIn" (which is called by Identify which is called by a bunch of bg processes) would basically reset ActiveDevice when the secret is stored and config.json points to a valid username.

**Comment by heronhaye** [comment]
Btw CI failures look legit. But probably test code needs to change, not app code. Just a guess

File: go/libkb/logout.go [hunk 2]
```diff
@@ -53,6 +53,13 @@ func (mctx MetaContext) LogoutUsernameWithOptions(username NormalizedUsername, o
 		return err
 	}
 
+	if options.KeepSecrets {
+		err = g.Env.GetConfigWriter().SetStayLoggedOut(true)
```
Comment (jakob223) [inline go/libkb/logout.go]: I think the reason TestBotSignup is failing is that we're calling this too often (when logout gets called by login, e.g.). Maybe we should call it from the engine instead? WDYT @patrickxb ?

**Comment by jakob223** [comment]
CI failures #1 are an unrelated KBFS test flake & KBFS tests timing out.

**Comment by jakob223** [comment]
This time it failed because Jenkins got bored.

**Pull request merged**